### PR TITLE
Add documentation examples for broken-noreturn, condition-evals-to-constant, and bad-thread-instantiation

### PR DESCRIPTION
**Pull request by clavedeluna**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |

| ✓   | :scroll: Docs          |

## Description

Add documentation examples for `broken-noreturn`, `condition-evals-to-constant`, and `bad-thread-instantiation`

Refs #5953

### CODE REVIEW

**Comment by clavedeluna** [comment]
> Create a PR. A changelog entry is not necessary :)

according to #5953

🤷‍♀️

File: doc/data/messages/c/condition-evals-to-constant/bad.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CONSTANT = 100
+bool(CONSTANT or True)  # [condition-evals-to-constant]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/condition-evals-to-constant/bad.py:2]:
```suggestion
def is_a_fruit(fruit):
    return bool(fruit in {"apple", "orange"} or True)  # [condition-evals-to-constant]
```

File: doc/data/messages/c/condition-evals-to-constant/good.py [hunk 2]
```diff
@@ -1 +1,2 @@
-# This is a placeholder for correct code for this message.
+CONSTANT = 100
+CONSTANT or True
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/condition-evals-to-constant/good.py:2]: This message probably means that there was a mistake so the "good" example can be different I think:
```suggestion
def is_a_fruit(fruit):
    return fruit in {"apple", "orange"}
```

File: doc/data/messages/b/broken-noreturn/bad.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from typing import NoReturn, Union
+
+
+def func2() -> Union[None, NoReturn]:  # [broken-noreturn]
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/broken-noreturn/bad.py:5]:
```suggestion
def exploding_apple(apple) -> Union[None, NoReturn]:  # [broken-noreturn]
    print(f"{apple} is about to explode")
```

File: doc/data/messages/b/broken-noreturn/good.py [hunk 4]
```diff
@@ -1 +1,5 @@
-# This is a placeholder for correct code for this message.
+from typing import NoReturn
+
+
+def func1() -> NoReturn:
+    raise Exception
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/broken-noreturn/good.py:5]:
```suggestion
def exploding_apple(apple) -> NoReturn:
    print(f"{apple} is about to explode")
    raise Exception("{apple} exploded !")
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you ! Some small suggestions to make the example more lively.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you 👍

**Pull request merged**